### ISSUE TITLE
Allow button release to trigger an action

### ISSUE THREAD

**Issue by jarpy** [WhiteMagic/JoystickGremlin#64]
Hopefully, I'm mistaken, but if not, I think this a regression. The ability to have different actions for press and release is extremely useful for toggle switches, such as those found on the Thrustmaster Warthog throttle.

**Comment by WhiteMagic** [comment]
Yeah that functionality was removed, for a few reasons, though since removing it there were a few usage cases in addition to yours that highlighted that there are actually quite a few people using this feature.

A few of the reasons I removed originally for were:
- the new container system made for some odd logic traps
- the interaction with the macro action made for very confusing consequences for press and release conditions
- it seemed only truly useful for remap buttons

So while it made some things easier, in the sense of implementation and covering most edge cases, it actually turns out that it enable quite a few usage scenarios that would be a shame to lose those. So I'm currently figuring out the best way of adding this back in. There is going to be a bit of thinking on both the UI and logic side of it but I'll find a way to add it back in.

**Comment by Carlton2001** [comment]
I cannot wait for the 9th release !
The Warthog throttle is not usable with the 8th because of the release action missing and the 7th don't offer the tempo and the chain...
I'm stuck with TARGET, help !

**Comment by WhiteMagic** [comment]
Yeah this whole thing took much longer then I hoped, work decided to actually be a lot of work. Though I've finally gotten a bit more time and just finished implementing the macro functions I wanted to add, so now I "only" have to test everything and fix the new bugs I've introduced. So sometime this month I should be done with R9.

**Comment by Carlton2001** [comment]
Nice, thanks a lot

**Comment by WhiteMagic** [comment]
Implemented new condition system with R9.

**Comment by jarpy** [comment]
Thank you, once again, @WhiteMagic. 💚

**Comment by izolini** [comment]
Could anyone show an example on how exactly to program a press/release switch on Warthog? I'm trying to do this but neither my Warthog or DIY panel (which has 16 switches) can be used this way. Although replicating examples that I saw on "Manual", I cannot make JGremlin recognize the "release" action, only "press". Tried with remap, macro, using conditions for joystick or action, whole container or single action, etc etc. Nothing seems to work. Currently using JGremlin v10. Read above that release 7 was working, where can I obtain this version, as this is the only resource (toggle between on/off) that I'm looking for?
